spring-boot-project/spring-boot-docker-compose/src/main/java/org/springframework/boot/docker/compose/service/connection/postgres/PostgresJdbcDockerComposeConnectionDetailsFactory.java

```diff
@@ -78,6 +80,42 @@ public String getJdbcUrl() {
 			return this.jdbcUrl;
 		}
 
+		private static final class PostgresqlJdbcUrlBuilder extends JdbcUrlBuilder {
+
+			private final Environment environment;
+
+			PostgresqlJdbcUrlBuilder(String driverProtocol, int containerPort, Environment environment) {
+				super(driverProtocol, containerPort);
+				this.environment = environment;
+			}
+
+			@Override
+			protected String getParameters(RunningService service) {
+				String parameters = super.getParameters(service);
+				return appendApplicationName(parameters);
+			}
+
+			private String appendApplicationName(String parameters) {
```

Comment:
This is quite involved. I wonder if there wouldn't be a way to simplify this. And if not, perhaps `getParameters` may need to evolve so that such a handling of the raw string is unnecessary.

Reply:
I will take a look at how it could be done better and easy to read. I have the same feeling that it looks quite messy.